client/lib/emails/can-add-mailboxes-to-email-subscription.ts

```diff
@@ -0,0 +1,20 @@
+import { canCurrentUserAddEmail } from 'calypso/lib/domains';
+import { hasEmailForwards } from 'calypso/lib/domains/email-forwarding';
+import { getGSuiteProductSlug, getGSuiteSubscriptionStatus } from 'calypso/lib/gsuite';
+import { getTitanProductSlug } from 'calypso/lib/titan';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Determines whether a domain has email provider and is everything set up to add new mailboxes
+ */
+
+export function canAddMailboxesToEmailSubscription( domain?: ResponseDomain ): boolean {
+	const emailIsProvisioning = ! getGSuiteProductSlug( domain ) && ! getTitanProductSlug( domain );
```

Comment:
```suggestion
	const emailSubscriptionIsProvisioned = Boolean( getGSuiteProductSlug( domain ) || getTitanProductSlug( domain ) || hasEmailForwards( domain ) );
```
Sorry for the back and forth, but this is probably even clearer.